category/applicative.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Instances for identity and composition functors
+-/
+
+import category.functor
+
+universe variables u v w u' v' w'
+
+section lemmas
+
+open function applicative
+
+variables {α β γ : Type u}
+variables {f : Type u → Type v}
+variables [applicative f] [is_lawful_applicative f]
+variables (g : β → γ)
+
+lemma pure_seq_pure {α β : Type u} (g : α → β) (x : α) :
```

Comment:
Why do you add this lemma? You don't need it, and it is solved by the simplifier anyway.